packages/hash/frontend/src/components/hooks/blockProtocolFunctions/useBlockProtocolDeleteLink.ts

```diff
@@ -0,0 +1,65 @@
+import { useMutation } from "@apollo/client";
+
+import { EmbedderGraphMessageCallbacks } from "@blockprotocol/graph";
+import { deleteLinkMutation } from "@hashintel/hash-shared/queries/link.queries";
+import { useCallback } from "react";
+import {
+  DeleteLinkMutation,
+  DeleteLinkMutationVariables,
+} from "../../../graphql/apiTypes.gen";
+import { parseLinkIdentifier } from "../../../lib/entities";
+
+export const useBlockProtocolDeleteLink = (): {
+  deleteLink: EmbedderGraphMessageCallbacks["deleteLink"];
+} => {
+  const [runDeleteLinkMutation] = useMutation<
+    DeleteLinkMutation,
+    DeleteLinkMutationVariables
+  >(deleteLinkMutation);
+
+  const deleteLink: EmbedderGraphMessageCallbacks["deleteLink"] = useCallback(
+    async ({ data }) => {
+      if (!data) {
+        return {
+          errors: [
+            {
+              code: "INVALID_INPUT",
+              message: "'data' must be provided for deleteLink",
+            },
+          ],
+        };
+      }
+
+      const { accountId: sourceAccountId, linkId } = parseLinkIdentifier(
+        data.linkId,
+      );
+
+      const { data: responseData } = await runDeleteLinkMutation({
+        variables: {
+          linkId,
+          sourceAccountId,
+        },
+      });
+
+      if (!responseData) {
+        return {
+          errors: [
+            {
+              code: "INVALID_INPUT",
+              message: "Error calling deleteLink",
+            },
+          ],
+        };
+      }
+
+      return {
+        data: true,
+      };
```

Comment:
Just curious, is there a reason we aren't using `void` as a return-type for BP methods? If there isn't, perhaps we should consider using `void` as the return type for `delete` methods in a future task, as to my understanding they can only return `true` and never return `false` as they would return errors instead.